### ISSUE TITLE
"Load function 'load' not found" when using Charts

### ISSUE THREAD

**Issue by martenson** [galaxyproject/galaxy#5738]
This is on dev at `2b8eb1ea778316e40584258558b8c284a080bb49` after `rm -rf client/node_modules` and `make client`.

I upload a sample bed file and visualize with charts. The list of viz is missing images and after clicking on top 3 from the list I get an alert and a JS console errors related to nvd3, but clicking on others also bring console errors(different).

![screenshot 2018-03-20 12 38 49](https://user-images.githubusercontent.com/1814954/37668972-13b8b234-2c3c-11e8-9227-83f8db25c161.png)
![screenshot 2018-03-20 12 38 59](https://user-images.githubusercontent.com/1814954/37668977-16cf762e-2c3c-11e8-89f8-bdebe773edb9.png)

![screenshot 2018-03-20 12 43 21](https://user-images.githubusercontent.com/1814954/37669142-77540942-2c3c-11e8-8203-77b946a94ebb.png)

**Comment by dannon** [comment]
(documenting chat from IRC) I think this is possibly uwsgi not serving the static content for viz's correctly on dev.

**Comment by nsoranzo** [comment]
This was also reported on gitter by @anuprulez : https://gitter.im/galaxyproject/Lobby?at=5ab0e1486f8b4b9946499544

**Comment by martenson** [comment]
Looks fully functional when I switched to Paste.

**Comment by dannon** [comment]
Awesome, hunch confirmed.  We need to set up the default uwsgi configuration with static mappings for these things.

**Comment by natefoo** [comment]
There's currently no static rule for viz/GIE plugins under uWSGI. This is an oversight that can be corrected for the "flat" viz/GIE plugins using the following config option in the `uwsgi:` section of `galaxy.yml`:

```
     route: ^/plugins/([^/]+)/([^/]+)/static/(.*)$ static:config/plugins/$1/$2/static/$3
```

However, that requires uWSGI internal routing support, which is not built in to uWSGI if you do not have PCRE development files installed at compile time, and is not built in to our current wheels. We can add it to our wheels (see galaxyproject/starforge#182), but if you get uWSGI from another source (system package manager or `pip install uwsgi` from PyPI) then routing support may not be available, and we won't be able to configure this route.

A "proper" solution to this problem would be for uWSGI to support the standard setuptools `extras_requires` specifiers so that we could require e.g. `uwsgi[pcre]==2.0.15`. This would have additional benefits in the long run because you can build uWSGI with libyaml support to get proper yaml parsing support, which would be preferable to the "quasi-yaml" parsing that is internal to uWSGI, and then we could additionally depend on that "extra" with `uwsgi[pcre,libyaml]==2.0.15`. But that'll require some uWSGI development and getting it upstreamed.

Additionally, the route above unfortunately doesn't work for the new "nested" viz plugins. To explain what I'm talking about, take a look at [config/plugins/visualizations](https://github.com/galaxyproject/galaxy/tree/2b8eb1ea778316e40584258558b8c284a080bb49/config/plugins/visualizations).

- "Flat" plugins like [graphviz](https://github.com/galaxyproject/galaxy/tree/2b8eb1ea778316e40584258558b8c284a080bb49/config/plugins/visualizations/graphviz) live directly in the `visualizations/` directory and have their static content at `config/plugins/visualizations/<plugin>/static/`.
- "Nested" plugins like [nvd3_bar](https://github.com/galaxyproject/galaxy/tree/2b8eb1ea778316e40584258558b8c284a080bb49/config/plugins/visualizations/nvd3/nvd3_bar) live under subdirectories of the `visualizations/` directory and are prepended with the subdirectory name and an underscore, so their static content is at `config/plugins/visualizations/<plug_dir>/<plug_dir>_<plugin>/static/`

For nginx we've [devised a mapping](https://github.com/galaxyproject/usegalaxy-playbook/blob/99b84394a10ef5ca04c1213b0f4c3bea76797d42/env/test/templates/nginx/galaxy_test.j2#L79) but it depends on nginx's `try_files` directive. There's probably an equivalent for Apache, and knowing uWSGI it's surely possible there too, but it becomes pretty complex.

I think maybe we need to take another look at plugin static content<sup>1</sup>. Perhaps as part of the client build process, it should be written to `/static`. Then no additional rules would be required<sup>2</sup>. Alternatively, it may be possible to set up these uWSGI static routes from inside the Galaxy application in a similar way to how they're done for Paste (which, I do not know how those are done), and then maybe it may not require internal routing support.

<sup>1. While we're at it, I don't think plugins belong in `config/` at all, really only the plugin configs (if configured by the admin) belong there.</sup>
<sup>2. Complex rules for viz/GIE static content are only required currently to avoid serving the full contents of the plugins dir, because `config/plugins/<plug_type>/<plugin>/config/` could contain sensitive data. If the static content was separated out in to `/static`, this would not be a problem.</sup>

**Comment by dannon** [comment]
Again crossposting from IRC:

Part of the intent behind the recent viz refactoring was to separate these things out into first class javascript packages with their own build infrastructure (parceljs, mostly), dependencies, resources, etc.  We can (should) use package.json dir/dist directives to indicate what to deploy to `/static`, and just crawl the directory structure as a part of the client build process, hopefully avoid the extra mappings/complexity completely.

**Comment by dannon** [comment]
I fixed this with the changes from #5810
